### DISCUSSION TITLE
Could you provide an example of how to use this library with react while initializing state with props?

### DISCUSSION THREAD

**Discussion by nratter** [pmndrs/valtio#336] [Q&A] [answered]
Could you provide an example of how to use this library with react while initializing state with props? Would it require the use of react context or is it possible without it?

Thanks!

**Comment by dai-shi** [comment, marked as answer]
If a proxy state is defined at module level, it's already initialized, so you should basically change it with `useEffect`.

```js
const state = proxy({ text: 'hello' })

const Component = ({ initialText }) => {
  useEffect(() => {
    state.text = initialText
  }, [])
  // ...
}
```

If this is not something you are looking for, a possible pattern is lazy initialization in render (which requires deeper understanding), or use context: https://github.com/pmndrs/valtio/wiki/How-to-use-with-context
